edgePy/DGEList.py

```diff
@@ -292,17 +292,40 @@ def rpkm(
         # self = self.cpm(log=log, prior_count=prior_count)
 
     def tpm(
-        self, transcripts: Mapping, log: bool = False, prior_count: float = PRIOR_COUNT
-    ) -> None:
-        """Return the DGEList normalized to reads per kilobase of transcript
-        length.
+        self,
+        gene_lengths: np.ndarray,
+        log: bool = False,
+        prior_count: float = PRIOR_COUNT,
+        mean_fragment_lengths: np.ndarray = None,
+    ) -> np.ndarray:
+        """Normalize the DGEList to transcripts per million
+
+        Args:
+            gene_lengths: vector of length for each gene in counts table
```

Comment:
Python doesn't really have a `vector` type in NumPy-speak. Maybe:

```
gene_lengths: 1D array of gene lengths for each gene in the rows of `DGEList.counts`.
mean_fragment_lengths: 1D array of mean fragment lengths for each sample in the columns of  `DGEList.counts` (optional)
```

Up to you.

Reply:
See earlier comment. I missed that one was a 'row' and the other a 'column'.